libsolidity/ast/AST.h

```diff
@@ -537,17 +537,19 @@ class FunctionDefinition: public CallableDeclaration, public Documented, public
 		bool _isConstructor,
 		ASTPointer<ASTString> const& _documentation,
 		ASTPointer<ParameterList> const& _parameters,
-		bool _isDeclaredConst,
 		std::vector<ASTPointer<ModifierInvocation>> const& _modifiers,
 		ASTPointer<ParameterList> const& _returnParameters,
+		bool _isView,
```

Comment:
If view, pure and payable are all mutually exclusive, the danger of having an inconsistent function type at any point is far worse than potentially hindering development.

Furthermore, we should always actively strive against any cargo cult thinking, so that these kinds of arguments are non-issues.